### ISSUE TITLE
Synch attributes title on GeoNode and MapStore

### ISSUE THREAD

**Issue by etj** [GeoNode/geonode#10378]
GeoNode allows editing titles and descriptions for attributes:

<details>
  <summary>screenshot</summary>

![image](https://user-images.githubusercontent.com/717359/206146875-22fe724d-a4db-4689-8d86-c453ec2e34ae.png)  
</details>

MapStore allows it as well (https://github.com/geosolutions-it/MapStore2-C040/issues/514):

<details>
  <summary>screenshot</summary>

![image](https://user-images.githubusercontent.com/717359/206147144-593e0b49-087c-4054-a3f0-dcba83d73c05.png)
</details>

GeoNode is going to include [#514](https://github.com/geosolutions-it/MapStore2-C040/issues/514) in this PR: https://github.com/GeoNode/geonode-mapstore-client/issues/1319

When it will be included, we'll have two different ways to edit titles and descriptions for the attributes, and they will be out of synch.

We need a way to keep in synch these labels.

**Comment by gannebamm** [comment]
There should be one way to edit those titles and descriptions. Since GeoNode did already provide this feature, I vote for using GeoNodes feature catalogue (the feature you show) as the single point of truth. MapStores table widget shall fetch this info.

Do I miss something?

**Comment by giohappy** [comment]
@gannebamm yes, this is the plan. At the moment MapStore retrieves the tables field names directly from Geoserver. The idea is to pass the attributes configuration from GeoNode.

**Comment by gannebamm** [comment]
@giohappy I just tested the v4 behaviour and realized that as a user you are not able to fetch the feature catalogue anywhere. The old template does not exist anymore and the attribute info is nowhere to be found.

**Comment by giohappy** [comment]
@gannebamm what do you mean by "feature catalog"? If you mean the attributes list, that's correct. You can see them when editing the resource metadata.
We could show the list of attributes in a tab, beside the "Info" tab, within the info panel (image below).

FYI the panel has been improved recently. Now it has support to include additional tabs by configuration. It's not documented yet, but now it should quite easy to include additional information.

![image](https://user-images.githubusercontent.com/571129/207831833-7cde9651-8083-4466-bb48-6ff49ac4cb53.png)

**Comment by gannebamm** [comment]
@giohappy yes the ISO feature catalogue was displayed in the metadata details:
![grafik](https://user-images.githubusercontent.com/6172324/208476386-6b1dcb44-efc5-49be-8d85-f11ee4acfba8.png)

You can download it:
![grafik](https://user-images.githubusercontent.com/6172324/208476239-a7946296-c501-4166-b28f-8e591ab78c13.png)

And it was part of the URL routes: https://atlas.thuenen.de/layers/agraratlas:geonode:levl_2003_lau2/feature_catalogue


We would like to get this info back in the details panel. The rework you have done to make it configurable sounds awesome! Maybe that could help in displaying multiple contact roles as per #10290

**Comment by gannebamm** [comment]
@giohappy This is the documentation about configuring the detail view, right? http://geonode.org/geonode-mapstore-client/master/module-DetailViewer.html

@mwallschlaeger this is most likely important for #10290 as it will need modifications on the detail panel.

**Comment by giohappy** [comment]
The attributes tab will be available in 4.1.0: https://github.com/GeoNode/geonode-mapstore-client/commit/09b6974382fa749aeab7ef5f2fed09464eeda689

**Comment by giohappy** [comment]
Closing as part of a repository cleanup: this issue was last updated before 2025-01-01. Not a judgment on the report itself — feel free to reopen or file a fresh issue against `main` if it's still relevant. Thanks for the report.
